### ISSUE TITLE
Portable version of Brave browser

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#694]
Carried over from https://github.com/brave/browser-laptop/issues/2953

> I had been looking for **Firefox** and later **Chromium** binaries so that I could place them on a USB stick\hard drive and move my browser with me (portability). However, I never found them.
> 
> I was pleasantly surprised to see **brave** raw binaries available for Linux.   
> 
> I would like to suggest the following:
> - Have binaries for all platforms available always - a USB stick\hard drive formatted using NTFS can store both Linux and Windows binaries and help the user achieve greater portability 
> - Make the binaries collection\archive\compressed file self-sufficient. That is, either they have no external dependencies or the dependencies are well-documented
> - If possible, make portable **brave** upgrade itself - [Tor browser (based on Firefox)](https://www.torproject.org/projects/torbrowser.html.en) already does this:
> 
> ![result](https://cloud.githubusercontent.com/assets/2631325/17423059/2eaa2684-5ad5-11e6-8cd9-bd1a44396268.gif)

**Comment by jawz101** [comment]
Or at least a custom installation.  My situation at work is I cannot install a program if it doesn't have a way to customize the installer.  Our GPO restricts editing of the desktop & start menu as those 2 things are standardized across the domain for end users and my user profile is also roaming.  For instance, with Firefox I can point my profile to a folder on my local machine as well as opt not to create icons.  A portable version would solve these things as well but addressing either a portable version or a custom installer with such options would solve either scenario.

TLDR; the installer is so basic people can use it portably or in an environment that is a little locked down.

**Comment by crazy-max** [comment]
I created a [portable release of brave 0.55.20](https://github.com/portapps/brave-portable/releases/tag/0.55.20-41). For the moment some restrictions do not allow to make the application completely portable. I explain it briefly on portapps/brave-portable#4. I will try to work on the file `pref_hash_store_impl` and add a switch to brave-core to get a full portable version. If some people can contribute to the brave-core repository to manage these problems I would be happy to participate.

**Comment by crazy-max** [comment]
Ok after applying a patch on Chromium and doing several tests on machines (Win7, 8.1, 10) I managed to set up portability. This allows the portability of everything, except certificates. I will create a PR on brave-core and a switch to enable/disable this behavior.

**Comment by adrianmak** [comment]
Opera has already been a China product

**Comment by Badum-Tiss** [comment]
Thank you so much, @crazy-max, for your valuable work. Your solution for Brave works flawlessly.

I hope your team keeps coming up with new releases.

**Comment by crazy-max** [comment]
@Badum-Tiss I'm a "one-man army" on Portapps but yeah I will follow the updates.

**Comment by Badum-Tiss** [comment]
@crazy-max I could help if and when there's a need.

I don't know much about Chromium's code however.
